jekyll/_cci2/install-machine-runner-3-on-docker.adoc

```diff
@@ -0,0 +1,110 @@
+
+The host needs to have Docker installed. Once the `runner` container is started, the container will immediately attempt to start running jobs. The container will be reused to run more jobs indefinitely until it is stopped.
+
+The number of containers running in parallel on the host is constrained by the host's available resources and your jobs' performance requirements.
+
+[#self-hosted-runner-terms-agreement]
+=== Self-hosted runner terms agreement
+
+{% include snippets/runner/terms.adoc %}
+
+[#create-namespace-and-resource-class]
+=== 1. Create namespace and resource class
+
+{% include snippets/runner/install-with-cli-steps.adoc %}
+
+=== 2. Create a Dockerfile that extends the machine runner 3.0 image
+
+Create a `Dockerfile.runner.extended` file. In this example, Python 3 is installed on top of the base image.
+
+```dockerfile
+FROM circleci/runner-agent:machine-3
+RUN sudo apt-get update; \
+    sudo apt-get install --no-install-recommends -y \
+        python3
+```
+
+[#build-the-docker-image]
+=== 3. Build the Docker image
+
+```shell
+docker build --file ./Dockerfile.runner.extended .
+```
+
+[#start-the-docker-container]
+=== 4. Start the Docker container
+
+NOTE: The environment variable values are not available to the `docker` command, so these environment variables are not visible in `ps` output.
+
+```shell
+CIRCLECI_RUNNER_NAME=<runner-name> CIRCLECI_RUNNER_API_AUTH_TOKEN=<runner-token> docker run --env CIRCLECI_RUNNER_NAME --env CIRCLECI_RUNNER_API_AUTH_TOKEN --name <container-name> <image-id-from-previous-step>
+```
+
+When the container starts, it will immediately attempt to start running jobs.
+
+[#start-the-docker-container-on-server]
+==== Start the Docker container on server
+
+When starting the Docker container on server, the `CIRCLECI_RUNNER_API_URL` environment variable will also need to be passed in using the `--env` flag.
+
+```shell
+CIRCLECI_RUNNER_NAME=<runner-name> CIRCLECI_RUNNER_API_AUTH_TOKEN=<runner-token> CIRCLECI_RUNNER_API_URL=<server_host_name> docker run --env CIRCLECI_RUNNER_NAME --env CIRCLECI_RUNNER_API_AUTH_TOKEN --env CIRCLECI_RUNNER_API_URL --name <container-name> <image-id-from-previous-step>
+```
+
+[#stopping-the-docker-container]
+=== Stopping the Docker container
+
+```shell
+docker stop <container-name>
+```
+
+[#remove-the-docker-container]
+=== Remove the Docker container
+
+In some cases you might need to fully remove a stopped machine runner container from the system, such as when recreating a container using the same name.
+
+```shell
+docker stop <container-name>; docker rm <container-name>;
+```
+
+[#migrating-from-launch-agent]
+=== Migrating from launch agent
+
+To migrate from launch agent to machine runner 3.0 on Docker simply requires stopping & removing the launch agent based containers and replacing them with the machine runner 3.0 based containers, as described above. The prefix of the name of environment variables has changed from `LAUNCH_AGENT_` to `CIRCLECI_RUNNER_`.
```

Comment:
```suggestion
NOTE: In machine runner 3.0, the environment variable name prefix has changed from `LAUNCH_AGENT_` to `CIRCLECI_RUNNER_`.

To migrate from launch agent to machine runner 3.0 on Docker, stop and remove the launch agent containers and replace them with machine runner 3.0 containers, using the commands described above. 
```

Reply:
Might be better to rework this as a step by step process but I will do that in a separate commit